### ISSUE TITLE
Big error for contacts without perm CiviCRM: profile listings and forms

### ISSUE THREAD

**Issue by eileenmcnaughton** [civicrm/org.civicrm.contactlayout#11]
We have some contacts in our system who do not have 'profile listings and forms' permission AND do not have a layout applied to them. They get the error below.

![screenshot 2018-08-30 12 26 22](https://user-images.githubusercontent.com/336308/44822594-eccc3580-ac4f-11e8-9745-289d682b8dae.png)

@colemanw I just gave them the above permission so it is no longer affecting us but I suspect some early returns would help performance too? Or perhaps just a judicious try-catch

**Comment by eileenmcnaughton** [comment]
@colemanw all in your imagination - look again :-)

**Comment by colemanw** [comment]
So users see this error even if there is no layout being displayed? That ain't right. I'll have a look.

**Comment by eileenmcnaughton** [comment]
yep -

**Comment by colemanw** [comment]
Ok, looks like there were 2 issues going on here.

1. There was a bug where layouts would be shown unconditionally to users who didn't belong to any group. That's why it appeared to be affecting users who weren't supposed to see any layout. I've fixed that with 13df736.
2. If the user doesn't have profile permissions and they *are* supposed to see a layout, the page blows up as in your screenshot above. I've fixed this with #12
3. That leaves the problem of clicking on the block to pull up the profile edit form. This still causes an ugly error. Solution for now is to grant profile permissions.
4. Did I say 2 issues? Apparently I can't count.
